File: python/src/data_structures/hashed/associative_array.py
# Наивная реализация ассоциативного массива


class Map[K, V]:
    def __init__(self) -> None:
        self.pairs = []

    def get(self, key: K) -> V | None:
        for pair in self.pairs:
            if pair.key == key:
                return pair.value
        return None  # Если пара не найдена, вернем null

    def set(self, key: K, value: V) -> None:
        for pair in self.pairs:
            if pair.key == key:
                pair.value = value
                return
        # Если пара с заданным ключом не найдена, добавим новую пару
        new_pair = Pair(key, value)
        self.pairs.append(new_pair)


class Pair[K, V]:
    def __init__(self, key: K, value: V) -> None:
        self.key = key
        self.value = value

File: go/data_structures/hashed/associative_array.go
package hashed

// Наивная реализация ассоциативного массива

type Pair struct {
	key   string
	value any
}

type Map struct {
	pairs []Pair
}

func (m *Map) Get(key string) any {
	for _, pair := range m.pairs {
		if pair.key == key {
			return pair.value
		}
	}
	return nil // Если пара не найдена, вернем null
}

func (m *Map) Set(key string, value any) {
	for i, pair := range m.pairs {
		if pair.key == key {
			m.pairs[i].value = value
			return
		}
	}
	// Если пара с заданным ключом не найдена, добавим новую пару
	newPair := Pair{key, value}
	m.pairs = append(m.pairs, newPair)
}
